### PR TITLE
[DPM-916] 4200 code on failed or closed session

### PR DESCRIPTION
**Pull request by georgiypetrov**

### CODE REVIEW

File: server/api/handlers/game_action.go [hunk 1]
```diff
@@ -31,6 +31,10 @@ func ProcessGameActionRequest(context context.Context, req *ws_interface.ApiRequ
 		return nil, ws_interface.NewHandlerError(ws_interface.InternalError, err)
 	}
 
+	if session.State == models.GameFailed {
```
Comment (algys) [inline server/api/handlers/game_action.go:34]: what if closed(Finished) ?

**Review by algys — CHANGES_REQUESTED** [review]

File: server/api/handlers/game_action.go (hunk 1 above)
Comment (georgiypetrov) [reply server/api/handlers/game_action.go:34]: if closed then state should be failed, shouldn't it? https://github.com/DaoCasino/game-contract-sdk/blob/master/sdk/include/game-contract-sdk/game_base.hpp#L526

Comment (justefg) [reply server/api/handlers/game_action.go:34]: does thi PR solve the problem?
@Satorit wanted 'session expired' to be displayed, looks like we won't
be able to do that if we merge it

Comment (justefg) [reply server/api/handlers/game_action.go:34]: I checked `game_base.hpp` looks like `game_failed` is only emitted in `close` action. 
Given how unlikely we are to rename it in the contract in my opinion we can assume that `game_failed = session expired`

Comment (algys) [reply server/api/handlers/game_action.go:34]: need to rename error to `SessionFailedOrFinished` and return it if `session.State == models.GameFailed || session.State == models.GameFinished`

**Review by algys — CHANGES_REQUESTED** [review]

File: server/api/handlers/game_action.go (hunk 1 above)
Comment (georgiypetrov) [reply server/api/handlers/game_action.go]: done

**Review by algys — APPROVED** [review]

**Pull request merged**